docs/source-pytorch/fabric/fabric.rst

```diff
@@ -389,6 +352,33 @@ To define your own behavior, subclass the relevant class and pass it in. Here's
     fabric = Fabric(plugins=[MyCluster()], ...)
 
 
+callbacks
+=========
+
+A callback class is a collection of methods that the training loop can call at a specific point in time, for example, at the end of an epoch.
+Add callbacks to Fabric to inject logic into your training loop from an external callback class.
+
+.. code-block:: python
+
+    class MyCallback:
+        def on_train_epoch_end(self, results):
+            ...
+
+You can then register this callback, or multiple ones directly in Fabric:
+
+.. code-block:: python
+
+    fabric = Fabric(callbacks=[MyCallback()]))
+
+
+Then, in your training loop, you can call a hook by its name. Any callback objects that have this hook will execute it:
+
+.. code-block:: python
+
+    # Call any hook by name
+    fabric.call("on_train_epoch_end", results={...})
```

Comment:
That means that all callbacks ideally support `**kwargs` to store undesired arguments for other callbacks right? Can we also pass positional arguments or just keyword args?

Reply:
Right now, all callbacks need to consistently accept the same args. That restriction is probably reasonable. We don't have special handling of args/kwargs right now

Reply:
In the doc I just added

"Pass in additional arguments that the hook requires"

Which should make it clear that if they are required, you need to pass them in. This implies your *args **kwargs handling needs to behave accordingly. 

We may want to inspect signatures and improve error messaging in the future